[HDInsight] Refactoring and bug fixes

wawon-msft opened the pull request:
There are 7 types of changes in this PR:
1. Fixing incorrect casing of properties to match the RP
2. Adding missing valid operation responses to extension operations
3. Reordered and put common parameters into the top-level parameters property
4. Removed inappropriate enum constraints for application, extension, and configuration names, which shouldn't originally have been enums
5. Moved SSH-related properties to the correct file
6. Removed update configuration API specific for gateway (the generic configuration update API remains)
7. Removed the capabilities API and added the usages API

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

wawon-msft committed:
Fixed inaccurate API specifications for HDInsight extensions: 1. typo in casing, 2. added missing valid response code

wawon-msft committed:
Minor refactoring: 1. Organized shared parameters (e.g. resource group), 2. Fixed inappropriate enum constraints, 3. Moved property definitions to appropriate file

AutorestCI commented:
# Automation for azure-sdk-for-go

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json:
```diff
@@ -66,15 +65,12 @@
                 "responses": {
+                    "200": {
+                        "description": "Ok response definition."
+                    },
```
```
error:  
{ code: 'RESPONSE_STATUS_CODE_NOT_IN_EXAMPLE',
  id: 'OAV111',
  message: 'Following response status codes "200" for operation "Extension_EnableMonitoring" were present in the swagger spec, however they were not present in x-ms-examples. Please provide them.',
  innerErrors: undefined }
error: Found errors in validating the response with statusCode "200" for x-ms-example "Enable cluster monitoring" in operation "Extension_EnableMonitoring".:
```

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json:
```diff
@@ -133,28 +122,24 @@
                 "responses": {
+                    "200": {
+                        "description": "Ok response definition."
+                    },
```
```
error:  
{ code: 'RESPONSE_STATUS_CODE_NOT_IN_EXAMPLE',
  id: 'OAV111',
  message: 'Following response status codes "200" for operation "Extension_DisableMonitoring" were present in the swagger spec, however they were not present in x-ms-examples. Please provide them.',
  innerErrors: undefined }
error: Found errors in validating the response with statusCode "200" for x-ms-example "Enable cluster monitoring" in operation "Extension_DisableMonitoring".:
```

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/applications.json:
```diff
@@ -141,33 +123,21 @@
+                    },
+                    {
+                        "$ref": "#/parameters/ApplicationNameParameter"
+                    },
```
```
{
  "id": "1016",
  "code": "ConstantStatusHasChanged",
  "message": "The 'constant' status changed from the old version to the new.",
  "jsonref": "#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.HDInsight~1clusters~1{clusterName}~1applications~1{applicationName}/put/applicationName",
  "json-path": "#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/applications/{applicationName}/put/applicationName",
  "type": "Error"
}
```

This is a breaking change in SDKs in each language.  The previous version had Enum with a single value "hue". The new one have a free form string. Relates to Put and Delete operations.

wawon-msft replied:
This should not have been an enum in the first place. The service accepted values other than "hue" before the enum constraint was introduced, and this would break them. Additionally, the values for application name are dynamic, so it wasn't appropriate to use enums for them.

hovsepm commented:
@wawon-msft please take a look to the breaking changes (they are a few). Do you have plans to upgrade API version in the nearest future? 

Ideally all these breaking changes should be parked till the next API version change.

wawon-msft commented:
@hovsepm These changes are breaking, but all of them are bug fixes. We have plans to upgrade the API version, but we would like to get the current spec to a good state before the fork.

wawon-msft committed:
Updated examples for extensions with valid response status code.

hovsepm commented:
@wawon-msft please add all bugfixes and changes that you mentioned over IM and ping me directly for the review.

wawon-msft committed:
Removed update configuration API specific for gateway

wawon-msft committed:
Removed the capabilities API, added the usages API

AutorestCI commented:
# Automation for azure-sdk-for-java

wawon-msft committed:
Reverted capabilities change and created new API version with the capabilities change

hovsepm commented:
@ravbhatnagar could you review the new API version 2018-06-01 for HDInsights in this PR?

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                "tags": [
+                    "Applications"
+                ],
+                "description": "Lists all of the applications HDInsight cluster.",
```
>Lists all of the applications HDInsight cluster. [](start = 32, length = 48)

Could you check this text string? It does not sound right.

wawon-msft replied:
Fixed grammar.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
```
please consider adding `default:` response to each endpoint that will describe error response message structures. [https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#default-response](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#default-response) . You can take a look to some examples in other swaggers e.g. [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json#L68](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json#L68)

wawon-msft replied:
This looks helpful. I'll look into this.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/ApplicationGetProperties"
```
>ApplicationGetProperties [](start = 51, length = 24)

The name for the object is confusing. I think `ApplicationProperties` (without Get) should be a better option.

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
```
Delete is marked as Long Running but Put is not. Could you verify that this is correct? My assumption is that since Delete is LRO then PUT should also be an LRO.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                    }
+                ],
+                "responses": {
+                    "200": {
```
Since this is an LRO - does it also support 202 status code?

wawon-msft replied:
Correct.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/cluster.json:
```diff
@@ -0,0 +1,1154 @@
+                        "description": "The cluster create request."
+                    }
+                ],
+                "responses": {
```
same comment about `default` response.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/cluster.json:
```diff
@@ -0,0 +1,1154 @@
+                    }
+                ],
+                "responses": {
+                    "200": {
```
>200 [](start = 21, length = 3)

Does this operation support 202 response?

wawon-msft replied:
Currently, only 200 is expected.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                "tags": [
+                    "Applications"
+                ],
+                "description": "Lists properties of the specified application.",
```
>Lists [](start = 32, length = 5)

the description message should start with "Gets" e.g. "Gets the specified application details/properties."

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                    "Applications"
+                ],
+                "description": "Lists all of the applications HDInsight cluster.",
+                "operationId": "Applications_List",
```
>Applications_List [](start = 32, length = 17)

the name of the operation should be `Applications_ListByCluster`

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/cluster.json:
```diff
@@ -0,0 +1,1154 @@
+                }
+            }
+        },
+        "errors": {
```
>errors [](start = 9, length = 6)

This type should be PascalCase

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/configurations.json:
```diff
@@ -0,0 +1,174 @@
+                        "description": "The cluster configurations."
+                    }
+                ],
+                "responses": {
```
same comment for this swagger - please consider Default response definition.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                        "$ref": "./examples/EnableLinuxClusterMonitoring.json"
+                    }
+                },
+                "operationId": "Extension_EnableMonitoring",
```
>Extension [](start = 32, length = 9)

should be plural - Extensions - i.e. `"operationId": "Extensions_EnableMonitoring"`

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                        "description": "The Operations Management Suite (OMS) workspace parameters."
+                    }
+                ],
+                "responses": {
```
please consider Default response for this swagger as well.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                        "$ref": "./examples/GetLinuxClusterMonitoringStatus.json"
+                    }
+                },
+                "operationId": "Extension_GetMonitoringStatus",
```
>Extension [](start = 32, length = 9)

Should be plural

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                        "$ref": "./examples/DisableLinuxClusterMonitoring.json"
+                    }
+                },
+                "operationId": "Extension_DisableMonitoring",
```
>Extension [](start = 32, length = 9)

Plural.

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                    "Extensions"
+                ],
+                "description": "Creates an HDInsight cluster extension.",
+                "operationId": "Extension_Create",
```
>Extension [](start = 32, length = 9)

Plural

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                ],
+                "responses": {
+                    "202": {
+                        "description": "OK. The request has succeeded."
```
202 -> Does this PUT an LRO? Does it have 200 return code as well?

wawon-msft replied:
This is a bug. enabling/disabling monitoring should match enabling/disabling extensions.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                        "$ref": "./examples/GetExtension.json"
+                    }
+                },
+                "operationId": "Extension_Get",
```
>Extension [](start = 32, length = 9)

Plural

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/locations.json:
```diff
@@ -0,0 +1,140 @@
+                    "Regions"
+                ],
+                "operationId": "Location_ListUsages",
+                "description": "Lists the usages for the specified location.",
```
I think this operation should be named as `Usages_ListByLocation` and the entire swagger should be renamed to "Usages.json" instead of Locations. Are you planning to add more operations in the nearest future for Location? e.g. put/get/delete ? if not then it does note make sense to me to have Location swagger that actually provides API for a child resource/operation of the Location.

wawon-msft replied:
We're most likely going to add more operations here.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/locations.json:
```diff
@@ -0,0 +1,140 @@
+                    "200": {
+                        "description": "OK response definition.",
+                        "schema": {
+                            "$ref": "#/definitions/UsagesResult"
```
>UsagesResult [](start = 51, length = 12)

the name should be rather `UsagesListResult`

wawon-msft replied:
Thanks, this change will also be in the 2015 version as well.

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/configurations.json:
```diff
@@ -0,0 +1,174 @@
+    "info": {
+        "title": "HDInsightManagementClient",
+        "description": "The HDInsight Management Client.",
+        "version": "2015-03-01-preview"
```
Whoops. I'll update this to "2018-06-01-preview" in each new file.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/locations.json:
```diff
@@ -0,0 +1,140 @@
+        }
+    },
+    "definitions": {
+        "LocalizedName": {
```
In the objects defined inside definition for this swagger there is not a single property that is sent to server so please mark every property as `"readOnly": true`

wawon-msft replied:
Sounds good.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/locations.json:
```diff
@@ -0,0 +1,140 @@
+    },
+    "paths": {
+        "/subscriptions/{subscriptionId}/providers/Microsoft.HDInsight/locations/{location}/usages": {
+            "get": {
```
does this operation support multi-page results? if so please add that functionality.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/operations.json:
```diff
@@ -0,0 +1,103 @@
+			}
+		}
+	},
+	"definitions": {
```
all the properties should be marked as `"readOnly": true`

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json:
```diff
@@ -0,0 +1,420 @@
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK response definition."
```
does this support 204?

wawon-msft replied:
This currently doesn't support 204.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json:
```diff
@@ -0,0 +1,420 @@
+                "tags": [
+                    "ScriptActions"
+                ],
+                "operationId": "ScriptActions_ListPersistedScripts",
```
>ScriptActions_ListPersistedScripts [](start = 32, length = 34)

I think it would be better to name this operation as "ScriptActions_ListPersistedScriptsByCluster"

hovsepm replied:
BTW, do we need to specify that this is PresistedScripts in the operation? Wouldn't it be enough to just call this operation as "ScriptActions_ListByCluster" ?

---
In reply to: [197284866](https://github.com/Azure/azure-rest-api-specs/pull/3256#discussion_r197284866) [](ancestors = 197284866)

wawon-msft replied:
I'll rename it the operation, but I'll keep it in the description. Persisted is an extraneous keyword.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json:
```diff
@@ -0,0 +1,420 @@
+                "tags": [
+                    "ScriptExecutionHistory"
+                ],
+                "operationId": "ScriptExecutionHistory_List",
```
>ScriptExecutionHistory_List [](start = 32, length = 27)

Operation name should be "ScriptExecutionHistory_ListByCluster"

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/examples/CreateApplication.json:
```diff
@@ -0,0 +1,61 @@
+                "name": "",
+                "type": "",
+                "etag": "",
+                "tags": {},
```
this examples are useless - please provide real values in the responses

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/applications.json:
```diff
@@ -425,6 +371,27 @@
             "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
         },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
this parameter should be specified on the method and not on the client - i.e. please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Fixed.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/applications.json:
```diff
@@ -425,6 +371,27 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/applications.json:
```diff
@@ -425,6 +371,27 @@
+            "description": "The name of the cluster."
+        },
+        "ApplicationNameParameter": {
+            "name": "applicationName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json:
```diff
@@ -1119,6 +1129,20 @@
             "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
         },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json:
```diff
@@ -1119,6 +1129,20 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/configurations.json:
```diff
@@ -115,136 +97,78 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/configurations.json:
```diff
@@ -115,136 +97,78 @@
+            "description": "The name of the cluster."
+        },
+        "ConfigurationNameParameter": {
+            "name": "configurationName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json:
```diff
@@ -376,19 +307,40 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json:
```diff
@@ -376,19 +307,40 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json:
```diff
@@ -376,19 +307,40 @@
+            "description": "The name of the cluster."
+        },
+        "ExtensionNameParameter": {
+            "name": "extensionName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:
```diff
@@ -258,6 +334,13 @@
             "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
         },
+        "LocationParameter": {
+            "name": "location",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/scriptActions.json:
```diff
@@ -436,6 +381,34 @@
             "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
         },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/scriptActions.json:
```diff
@@ -436,6 +381,34 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/scriptActions.json:
```diff
@@ -436,6 +381,34 @@
+            "description": "The name of the cluster."
+        },
+        "ScriptNameParameter": {
+            "name": "scriptName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/scriptActions.json:
```diff
@@ -436,6 +381,34 @@
+            "description": "The name of the script."
+        },
+        "ScriptExecutionIdParameter": {
+            "name": "scriptExecutionId",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+            "description": "The name of the cluster."
+        },
+        "ApplicationNameParameter": {
+            "name": "applicationName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/cluster.json:
```diff
@@ -0,0 +1,1154 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/cluster.json:
```diff
@@ -0,0 +1,1154 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/configurations.json:
```diff
@@ -0,0 +1,174 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/configurations.json:
```diff
@@ -0,0 +1,174 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/configurations.json:
```diff
@@ -0,0 +1,174 @@
+            "description": "The name of the cluster."
+        },
+        "ConfigurationNameParameter": {
+            "name": "configurationName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+            "description": "The name of the cluster."
+        },
+        "ExtensionNameParameter": {
+            "name": "extensionName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
**Done.**

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/locations.json:
```diff
@@ -0,0 +1,140 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "LocationParameter": {
+            "name": "location",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json:
```diff
@@ -0,0 +1,420 @@
+            "description": "The subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+        },
+        "ResourceGroupNameParameter": {
+            "name": "resourceGroupName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json:
```diff
@@ -0,0 +1,420 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json:
```diff
@@ -0,0 +1,420 @@
+            "description": "The name of the cluster."
+        },
+        "ScriptNameParameter": {
+            "name": "scriptName",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json:
```diff
@@ -0,0 +1,420 @@
+            "description": "The name of the script."
+        },
+        "ScriptExecutionIdParameter": {
+            "name": "scriptExecutionId",
```
please add `"x-ms-parameter-location": "method"` property.

wawon-msft replied:
Done.

hovsepm commented:
Please fix all following discrepancy errors:
```
[ 'specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/capabilities.json' ]
error:  
{ code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
  id: 'OAV112',
  message: 'Response statusCode "200" for operation "Extension_Create" is provided in exampleResponseValue, however it is not present in the swagger spec.',
  innerErrors: undefined }
error: Found errors in validating the response with statusCode "200" for x-ms-example "Create a monitoring extension on Hadoop Linux cluster" in operation "Extension_Create".:
 
{ code: 'RESPONSE_VALIDATION_ERROR',
  id: 'OAV108',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "Create a monitoring extension on Hadoop Linux cluster" in operation "Extension_Create".',
  innerErrors: 
   [ { code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
       id: 'OAV112',
       message: 'Response statusCode "200" for operation "Extension_Create" is provided in exampleResponseValue, however it is not present in the swagger spec.',
       innerErrors: undefined } ] }
error:  
{ code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
  id: 'OAV112',
  message: 'Response statusCode "200" for operation "Extension_Delete" is provided in exampleResponseValue, however it is not present in the swagger spec.',
  innerErrors: undefined }
error: Found errors in validating the response with statusCode "200" for x-ms-example "Delete an extension" in operation "Extension_Delete".:
 
{ code: 'RESPONSE_VALIDATION_ERROR',
  id: 'OAV108',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "Delete an extension" in operation "Extension_Delete".',
  innerErrors: 
   [ { code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
       id: 'OAV112',
       message: 'Response statusCode "200" for operation "Extension_Delete" is provided in exampleResponseValue, however it is not present in the swagger spec.',
       innerErrors: undefined } ] }
{ AssertionError [ERR_ASSERTION]: swagger "specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/extensions.json" contains model validation errors.
    at Context.<anonymous> (/home/travis/build/Azure/azure-rest-api-specs/test/model.js:20:16)
    at <anonymous>
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '==' }
error:  
{ code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
  id: 'OAV112',
  message: 'Response statusCode "200" for operation "Extension_Create" is provided in exampleResponseValue, however it is not present in the swagger spec.',
  innerErrors: undefined }
error: Found errors in validating the response with statusCode "200" for x-ms-example "Create a monitoring extension on Hadoop Linux cluster" in operation "Extension_Create".:
 
{ code: 'RESPONSE_VALIDATION_ERROR',
  id: 'OAV108',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "Create a monitoring extension on Hadoop Linux cluster" in operation "Extension_Create".',
  innerErrors: 
   [ { code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
       id: 'OAV112',
       message: 'Response statusCode "200" for operation "Extension_Create" is provided in exampleResponseValue, however it is not present in the swagger spec.',
       innerErrors: undefined } ] }
error:  
{ code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
  id: 'OAV112',
  message: 'Response statusCode "200" for operation "Extension_Delete" is provided in exampleResponseValue, however it is not present in the swagger spec.',
  innerErrors: undefined }
error: Found errors in validating the response with statusCode "200" for x-ms-example "Delete an extension" in operation "Extension_Delete".:
 
{ code: 'RESPONSE_VALIDATION_ERROR',
  id: 'OAV108',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "Delete an extension" in operation "Extension_Delete".',
  innerErrors: 
   [ { code: 'RESPONSE_STATUS_CODE_NOT_IN_SPEC',
       id: 'OAV112',
       message: 'Response statusCode "200" for operation "Extension_Delete" is provided in exampleResponseValue, however it is not present in the swagger spec.',
       innerErrors: undefined } ] }
{ AssertionError [ERR_ASSERTION]: swagger "specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json" contains model validation errors.
    at Context.<anonymous> (/home/travis/build/Azure/azure-rest-api-specs/test/model.js:20:16)
    at <anonymous>
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '==' }
```

hovsepm reviewed (changes requested):
Please address review comments.

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                    }
+                ],
+                "responses": {
+                    "202": {
```
This currently doesn't support 204.

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/cluster.json:
```diff
@@ -0,0 +1,1154 @@
+                ],
+                "responses": {
+                    "202": {
+                        "description": "Accepted response definition."
```
This currently doesn't support 204.

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK. The request has succeeded."
```
This currently doesn't support 204.

wawon-msft committed:
Various minor changes: fixed API version in specs and examples, renamed properties, renamed operations, fixed examples, added x-ms-parameter-location to certain parameters, and made usages readonly.

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/configurations.json:
```diff
@@ -115,136 +97,78 @@
+            "description": "The name of the resource group."
+        },
+        "ClusterNameParameter": {
+            "name": "clusterName",
```
Done.

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -1,403 +1,408 @@
-{
```
Not quite sure what happened with the diff. The changes here were:
- Updated version
- Changed Applications_List to Applications_ListByCluster
- Changed description of Applications_ListByCluster
- Changed ApplicationGetProperties to ApplicationProperties
- Changed casing of Errors
- Added x-ms-parameter-location to parameters

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/applications.json:
```diff
@@ -0,0 +1,403 @@
+    "info": {
+        "title": "HDInsightManagementClient",
+        "description": "The HDInsight Management Client.",
+        "version": "2015-03-01-preview"
```
My bad, this is fixed.

wawon-msft committed:
Fixed bug for applications where expected response was missing

wawon-msft committed:
Renamed operation that lists script actions

wawon-msft committed:
Fixed examples for applications to include expected status code from bug fix

wawon-msft commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,346 @@
+                ],
+                "responses": {
+                    "202": {
+                        "description": "OK response definition."
```
This was a bug. I've updated this.

wawon-msft committed:
Fixed inconsistent description for Operation in cluster.json

hovsepm commented:
@wawon-msft please fix following linter errors (FYI model names should be PascalCase while property names should be camelCase)
```
Executing: npx autorest@2.0.4152 --validation --azure-validator --message-format=json specification/hdinsight/resource-manager/readme.md --tag=package-2015-03-preview
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'FilterMode', for definition: 'vmSizeCompatibilityFilter' must follow camelCase style. Example: 'filterMode'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:166:16 ($.definitions.vmSizeCompatibilityFilter.properties.FilterMode)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:166:16 ($.definitions.vmSizeCompatibilityFilter.properties.FilterMode)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:166:16 ($.definitions.vmSizeCompatibilityFilter.properties.FilterMode)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'Regions', for definition: 'vmSizeCompatibilityFilter' must follow camelCase style. Example: 'regions'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:170:16 ($.definitions.vmSizeCompatibilityFilter.properties.Regions)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:170:16 ($.definitions.vmSizeCompatibilityFilter.properties.Regions)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:170:16 ($.definitions.vmSizeCompatibilityFilter.properties.Regions)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'ClusterFlavors', for definition: 'vmSizeCompatibilityFilter' must follow camelCase style. Example: 'clusterFlavors'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:177:16 ($.definitions.vmSizeCompatibilityFilter.properties.ClusterFlavors)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:177:16 ($.definitions.vmSizeCompatibilityFilter.properties.ClusterFlavors)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:177:16 ($.definitions.vmSizeCompatibilityFilter.properties.ClusterFlavors)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'NodeTypes', for definition: 'vmSizeCompatibilityFilter' must follow camelCase style. Example: 'nodeTypes'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:184:16 ($.definitions.vmSizeCompatibilityFilter.properties.NodeTypes)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:184:16 ($.definitions.vmSizeCompatibilityFilter.properties.NodeTypes)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:184:16 ($.definitions.vmSizeCompatibilityFilter.properties.NodeTypes)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'ClusterVersions', for definition: 'vmSizeCompatibilityFilter' must follow camelCase style. Example: 'clusterVersions'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:191:16 ($.definitions.vmSizeCompatibilityFilter.properties.ClusterVersions)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:191:16 ($.definitions.vmSizeCompatibilityFilter.properties.ClusterVersions)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:191:16 ($.definitions.vmSizeCompatibilityFilter.properties.ClusterVersions)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'region_name', for definition: 'regionalQuotaCapability' must follow camelCase style. Example: 'region_name'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:210:16 ($.definitions.regionalQuotaCapability.properties.region_name)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:210:16 ($.definitions.regionalQuotaCapability.properties.region_name)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:210:16 ($.definitions.regionalQuotaCapability.properties.region_name)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'cores_used', for definition: 'regionalQuotaCapability' must follow camelCase style. Example: 'cores_used'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:214:16 ($.definitions.regionalQuotaCapability.properties.cores_used)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:214:16 ($.definitions.regionalQuotaCapability.properties.cores_used)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:214:16 ($.definitions.regionalQuotaCapability.properties.cores_used)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'cores_available', for definition: 'regionalQuotaCapability' must follow camelCase style. Example: 'cores_available'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:219:16 ($.definitions.regionalQuotaCapability.properties.cores_available)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:219:16 ($.definitions.regionalQuotaCapability.properties.cores_available)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:219:16 ($.definitions.regionalQuotaCapability.properties.cores_available)"
}
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'vmSize_filters', for definition: 'capabilitiesResult' must follow camelCase style. Example: 'vmSize_filters'.",
  "id": "R3016",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:262:16 ($.definitions.capabilitiesResult.properties.vmSize_filters)"
  ],
  "jsonref": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:262:16 ($.definitions.capabilitiesResult.properties.vmSize_filters)",
  "json-path": "file:///home/travis/build/Azure/azure-rest-api-specs/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json:262:16 ($.definitions.capabilitiesResult.properties.vmSize_filters)"
}
```

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json:
```diff
@@ -0,0 +1,357 @@
+                    "Extensions"
+                ],
+                "description": "Creates an HDInsight cluster extension.",
+                "operationId": "Extensions_Create",
```
>"operationId": "Extensions_Create", [](start = 16, length = 35)

if the Update of the extension is allowed and it overwrites the existing one - then this operationId should be `Extensions_CreateOrUpdate`

wawon-msft replied:
We don't allow updates on the extension.

hovsepm commented on specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/locations.json:
```diff
@@ -0,0 +1,144 @@
+                "tags": [
+                    "Regions"
+                ],
+                "operationId": "Location_ListUsages",
```
>"operationId": "Location_ListUsages [](start = 16, length = 35)

`Location` should be plural word here.

wawon-msft replied:
Good catch.

wawon-msft committed:
Removed unnecessary definition, added default responses on all endpoints

wawon-msft committed:
Fixed PascalCasing of properties, and made Locations plural in operation IDs

wawon-msft commented:
@hovsepm: Those linter errors were partially why we introduced the new API version. The property names had underscores in them, which were not camelcase to the linter. I renamed the file to locations.json, which might have raised "new" errors, but these are resolved with the breaking change (by removing the capabilities API).

hovsepm commented:
@wawon-msft looks good to me. Once ARM team will review and sign-off you PR I'll merge it.

wawon-msft commented:
@hovsepm @ravbhatnagar Thanks so much, guys!
